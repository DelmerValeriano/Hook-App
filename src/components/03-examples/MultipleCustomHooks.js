import React  from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import '../02-useEffect/effects.css';

export const MultipleCustomHooks = () => {


    const {counter,increment} =useCounter(1);

    const {loading,data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    const {author,quote} = !!data && data[0];

  

    return (
    <div>
        <h1 className="text-center">Breaking Quotes</h1>
        <hr/>

        {/* //ternario */}
        {
            loading
            ?
                (
                    <div className="alert alert-success text-center">
                        Loading...
                    </div>
                )
            :
                (
                    <blockquote className="blockquote text-end">
                        <p className="mb-3">{quote}</p>
                        <footer className="blockquote-footer">{author}</footer>
                   </blockquote>
                )
        }

       
       
        <button className="btn btn-primary" onClick={increment}>
            Siguiente quote
        </button>


     
    </div>
  )
}
