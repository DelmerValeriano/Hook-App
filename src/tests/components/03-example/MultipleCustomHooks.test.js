
    import React from 'react'
    import {shallow} from 'enzyme'
    import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
    import { useFetch } from '../../../hooks/useFetch';
    import { useCounter } from '../../../hooks/useCounter';

    jest.mock('../../../hooks/useFetch');
    jest.mock('../../../hooks/useCounter');

    useCounter

    describe('prueba en MultipleCustomHooks', () => {  


        useCounter.mockReturnValue({
            counter:10,
            increment: ()=>{}
        })


        test('debe de mostrarse correctamente', () => {  
            useFetch.mockReturnValue({
                data: false,
                loading: true, 
                error: null,
            })

            const wrapper = shallow(<MultipleCustomHooks/>);
            expect(wrapper).toMatchSnapshot();

        });

        test('debe de mostrar la informacion', () => {  

            useFetch.mockReturnValue({
                data: [{author:'Delmer',quote:'Hola mundo'}],
                loading: false, 
                error: null,
            });
            const wrapper = shallow(<MultipleCustomHooks/>);
            expect(wrapper.find('.alert').exists()).toBe(false);
            expect(wrapper.find('.mb-3').text().trim()).toBe('Hola mundo');
            expect(wrapper.find('footer').text().trim()).toBe('Delmer');


        })
    })
