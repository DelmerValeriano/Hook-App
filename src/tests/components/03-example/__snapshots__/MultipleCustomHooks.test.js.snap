// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en MultipleCustomHooks debe de mostrarse correctamente 1`] = `
<div>
  <h1
    className="text-center"
  >
    Breaking Quotes
  </h1>
  <hr />
  <div
    className="alert alert-success text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
